src/components/Modal/index.js

```diff
@@ -1,139 +1,72 @@
-    mountEntry: 'modal-root',
-  }
-
-  componentDidMount () {
-    document.addEventListener('keydown', this.onPressEsc)
-  }
+    className: PropTypes.string,
+    show: PropTypes.bool,
 
-  componentWillUnmount () {
-    document.removeEventListener('keydown', this.onPressEsc)
+    onCloseClick: PropTypes.func,
   }
 
-  onPressEsc = (event) => {
-    const { onClose, shouldCloseOnEsc } = this.props
-    const escKey = 27
-    const escKeyWasPressed = event.keyCode === escKey
-    if (shouldCloseOnEsc && escKeyWasPressed) {
-      onClose()
+  onKeyDown = (event) => {
+    if (event.keyCode === 27) {
+      this.close()
     }
   }
 
-  onClickOutside = () => {
-    const { onClose, shouldCloseOnClickOutside } = this.props
-    if (shouldCloseOnClickOutside) {
-      onClose()
+  close = () => {
+    const {
+      onCloseClick,
+    } = this.props
+
+    if (onCloseClick) {
+      onCloseClick()
     }
   }
 
-  container = React.createRef();
-
   render () {
     const {
-      isOpen,
-      onClose,
       children,
       className,
-      HeaderComponent,
-      closeButtonPosition,
-      mountEntry,
-      ...props
+      show,
     } = this.props
-    const cnRoot = 'mc-components-modal'
-
-    const classNames = cn(
-      `${cnRoot}__container`,
-      { [className]: Boolean(className) },
-    )
-    const cnClose = cn(
-      `${cnRoot}__close-icon`,
-      `${cnRoot}__close-icon--${closeButtonPosition}`,
-    )
 
-    if (!isOpen) {
-      return false
+    if (!show) {
+      return null
     }
 
-    return (
-      <ModalPortal mountEntry={mountEntry}>
-        { /outside/.test(closeButtonPosition) &&
-          <Close
-            className={cnClose}
-            onClick={onClose}
-          /> }
-        <ClickOutside
-          divRef={this.container}
-          onClickOutside={this.onClickOutside}
+    const classes = cn({
+      [className]: className,
+      'mc-modal': true,
+    })
+
+    return createPortal(
+      <div
+        className={classes}
+        onKeyDown={this.onKeyDown}
+      >
+        <a
+          className='mc-modal__backdrop'
+          onClick={this.close}
         >
-          <div ref={this.container} className={classNames}>
-            <RoundedBox
-              className={`${cnRoot}__rounded-box`}
-              HeaderComponent={HeaderComponent}
-              {...props}
-            >
-              <Close
-                className={cn([
-                  `${cnRoot}__inner-close-icon`,
-                  `${cnRoot}__inner-close-icon--${closeButtonPosition}`,
-                  {
-                    [`${cnRoot}__inner-close-icon--desktop-active`]: /inside/.test(closeButtonPosition),
-                  },
-                ])}
-                color='#000'
-                onClick={onClose}
-              />
-              {children}
-            </RoundedBox>
-          </div>
-        </ClickOutside>
-      </ModalPortal>
+          <IconClose className='mc-modal__close' />
```

Comment:
Modals don't always have an 'x' in the overlay, some of them have the 'x' inside the white rounded box and some don't have an 'x' at all

Reply:
btw I think most of the modals that are implemented have the 'x' inside the modal.

Example:
<img width="543" alt="screen shot 2018-10-10 at 6 46 58 pm" src="https://user-images.githubusercontent.com/18464392/46767912-e30d0780-ccbc-11e8-9fe6-373db79c85b2.png">

Reply:
I knew this one would come up.

So, with stripping out unnecessary parts of a modal, I battled with this one the most...but here's where I landed.  If the Modal component is opinionated on anything, it should be a consistent user experience (even if not a consistent design).  Where the close button exists should be something we can draw a line on.  Additionally, this is the least invasive place to put the close button for a designer.  If we cannot unify where a close button on a modal exists, we shouldn't be building a component library.

Reply:
I think we will need the close button inside the modal for the mobile view. All the mobile design modals that I saw have the close button inside, just because there is no space in a mobile view to put it outside.

Reply:
i have the X hovering atop the content, so in the case that a modal is too large the close button is still displayed.  i can run this by design